### ISSUE TITLE
MacBook M3 pro build failing

### ISSUE THREAD

**Issue by Caroljpeg** [pierotofy/OpenSplat#141]
Hello, I'm trying to run OpenSplat on a MacBook pro M3 pro (Sequoia 15.1.1). The building process was pretty straightforward and completed without showing any error, but when trying to run `./opensplat --help` I get this error `Bus error: 10`

**Comment by pierotofy** [comment]
Mm, strange; I just recompiled it on my MacBook Air M1 (Sonoma 14.6.1) and it seemed to work.

**Comment by AsherJingkongChen** [comment]
There are other 3DGS implementations work fine on Apple GPUs, you can look for them, too.

**Comment by kaicherry** [comment]
Hi...
I use opensplat on  silicon..daily now; it's actually quite nice and I wish @pierotofy  didn't think it can't "compete against VC stuff" - I am using it *because* the "VC Stuff" is going to become "prohibitive" (it's part of the model, I am "around Silicon Valley"...)

Anyway, @pierotofy, this error *does* 'pop up' when I re-compile, from time to time, and I believe it is due to conflicts/libs being picked up from `conda` libs and not the `brew` ones, if this in any way helps.

Also..OpenSplat is great :D 
I don't what "other 3DGS implementations work fine on Apple GPUs"...if at all :)
-K

**Comment by jared-krauss** [comment]
@kaicherry would you mind having a look at this and telling me where I'm going wrong? haha struggling to get the banana .ply made on my M1 Pro. 

https://docs.google.com/document/d/1Qfyar0fL89X2R9JqByIpli1LtgETBIXKnbNzIyfjHx4/edit?usp=sharing

**Comment by LordLKY** [comment]
hi, i confront the same “bus error” problem on Mac mini M2 (Sequoia 15.1.1).  has anyone figured out the reason?

**Comment by kaicherry** [comment]
Yes. When you update macOS, the `.dylibs` change and you need to build `opensplat` again. :/
NB: You should follow the `BUILD.md` -however- download and unzip `libpytorch-2.5.1` from [https://download.pytorch.org/libtorch/cpu/](https://download.pytorch.org/libtorch/cpu/) (build errors on 2.6.0)

@jared-krauss  WOW! You have added a TON of complexity; all you need to do is `brew  install colmap`.  That's 'stage one' and is all that is needed/'the minimum' to generate sparse point clouds that work with opensplat on our Macs.

`OpenMVS` I was able to compile following the build directions.

The lack of "density" from the `Densify` stuff was a fundamental lack of understanding on my part, and apparently others, too, on *how* to get the dense point cloud "back into" the 'sparse' point cloud that `opensplat` reads.

**Comment by jared-krauss** [comment]
@kaicherry i was following you until the lack of density part. referring to needs OpenMVS to get the dense point cloud?

Is this necessar for GS, or just for using it in other 3d programs so one has volumetric data as well? 

Sorry if this is basic question. Lol. I studied history in university haha, not this stuff.

**Comment by jared-krauss** [comment]
I finally got my first one to work! Thanks for the advice on the redownload. Also, who knew, but I wasn't putting my images in an "images" folder alongside colmap builds, so the opensplat couldn't read it

**Comment by szpnygo** [comment]
> 我终于让第一个工作了！感谢您关于重新下载的建议。另外，谁知道呢，但我没有将我的图像放在 colmap 构建旁边的“图像”文件夹中，因此 opensplat 无法读取它

how？
